lib/realtime_web/controllers/broadcast_controller.ex

```diff
@@ -30,26 +33,45 @@ defmodule RealtimeWeb.BroadcastController do
     responses: %{
       202 => EmptyResponse.response(),
       403 => EmptyResponse.response(),
-      422 => UnprocessableEntityResponse.response()
+      422 => UnprocessableEntityResponse.response(),
+      429 => TooManyRequestsResponse.response()
     }
   )
 
   def broadcast(%{assigns: %{tenant: tenant}} = conn, attrs) do
     with %Ecto.Changeset{valid?: true} = changeset <-
            BatchBroadcast.changeset(%BatchBroadcast{}, attrs),
          %Ecto.Changeset{changes: %{messages: messages}} <- changeset,
-         events_per_second_key <- Tenants.events_per_second_key(tenant) do
+         events_per_second_key <- Tenants.events_per_second_key(tenant),
+         :ok <- check_rate_limit(events_per_second_key, tenant, length(messages)) do
       for %{changes: %{topic: sub_topic, payload: payload, event: event}} <- messages do
         tenant_topic = Tenants.tenant_topic(tenant, sub_topic)
         payload = %{"payload" => Map.merge(payload, %{"event" => event})}
 
         Endpoint.broadcast_from(self(), tenant_topic, "broadcast", payload)
 
         GenCounter.add(events_per_second_key)
-        Logger.info("Broadcasted message to using HTTP API: " <> tenant_topic)
+        Logger.info("Broadcasted message using HTTP API to topic " <> tenant_topic)
```

Comment:
Thought you were only adding this for testing. We need to pull it out we don't want every Broadcast request being logged here.